storage/entities/base_entity.js

```diff
@@ -305,33 +305,43 @@ class BaseEntity {
 	parseFilters(filters) {
 		let filterString = null;
 
-		const parseFilterObject = object =>
-			Object.keys(object)
-				.map(key => this.filters[key])
-				.join(' AND ');
-
 		if (Array.isArray(filters)) {
-			filterString = filters
-				.map(filterObject => parseFilterObject(filterObject))
-				.join(' OR ');
-		} else if (typeof filters === 'object') {
-			filterString = parseFilterObject(filters);
+			filterString = this._parseArrayFilter(filters);
+		} else if (Object.keys(filters).length > 0) {
+			filterString = this._parseFilterObject(filters);
 		}
 
-		let filtersObject = filters;
+		const filtersObject = BaseEntity._filtersToObject(filters);
 
-		if (Array.isArray(filters)) {
-			filtersObject = Object.assign({}, ...filters);
-		}
+		return this._filterClause(filterString, filtersObject);
+	}
 
+	_filterClause(filterString, filtersObject) {
+		let filterClause;
 		if (filterString) {
-			return `WHERE ${this.adapter.parseQueryComponent(
+			filterClause = `WHERE ${this.adapter.parseQueryComponent(
 				filterString,
 				filtersObject
 			)}`;
 		}
 
-		return '';
+		return filterClause || '';
+	}
+
+	static _filtersToObject(filters) {
```

Comment:
Keep it non-static and add es-lint check.

Reply:
if it's not using state, why not just make it to normal function?

Reply:
I'm resolving this for the PR to move forward let's discuss on implementing a library exporting functions in the issue: https://github.com/LiskHQ/lisk/issues/2669 as that issue is geared towards architectural/organisational changes.